### PR TITLE
[INLONG-5314][TubeMQ] Reconnect to another address if heartbeat encounters exception

### LINKED ISSUE #5314: [Bug][TubeMQ] When a non-master node is configured in the first position of the address, the service cannot be started
### What happened

When a non-master node is configured in the first position of the address, the service cannot be started.

### What you expected to happen

The master-slave order of the client relationship is not required when configuring addresses.

### How to reproduce

When a non-master node is configured in the first position of the address, the service cannot be started. 

The client program panics when the server performs a master-slave switch

### Environment

centos 7

github.com/apache/inlong/inlong-tubemq/tubemq-client-twins/tubemq-client-go v0.0.0-20220731125344-27b1a02fc0db

### InLong version

master

### InLong Component

InLong TubeMQ

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TszKitLo40**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5314 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*
Reconnect to another address if heartbeat encounters exception

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go [hunk 1]
```diff
@@ -99,36 +99,43 @@ func (h *heartbeatManager) consumerHB2Master() {
...
+			log.Warnf("[CONSUMER] hb2master found no-node or standby, re-register, client=%s", h.consumer.clientID)
+			address := h.consumer.master.Address
+			go func() {
+				err := h.consumer.register2Master(rsp.GetErrCode() != errs.RetErrHBNoNode)
+				if err != nil {
 					return
 				}
-				log.Warnf("[CONSUMER] heartBeat2Master failure to (%s) : %s, client=%s", h.consumer.master.Address, rsp.GetErrMsg(), h.consumer.clientID)
+			}()
+			if rsp != nil && rsp.GetErrCode() != errs.RetErrHBNoNode {
```
Comment (charlely) [inline inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:125]: Always cann't run here.

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go [hunk 2]
```diff
@@ -99,36 +99,43 @@ func (h *heartbeatManager) consumerHB2Master() {
...
+				h.mu.Lock()
```
Comment (charlely) [inline inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:126]: L126-L132 After you can package a heartbeats struct, support Incr Add function to manage heartbeats.

**Review by mrcat2018 — APPROVED** [review]

**Review by charlely — APPROVED** [review]
LGTM

**Comment by TszKitLo40** [comment]
@gosonzhang Maybe we can test this PR first.

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go [hunk 3]
```diff
@@ -99,36 +99,45 @@ func (h *heartbeatManager) consumerHB2Master() {
 	}
 
 	rsp, err := h.sendHeartbeatC2M(m)
-	if err != nil {
-		log.Errorf("consumer hb err %s", err.Error())
+	if err != nil || !rsp.GetSuccess() {
```
Comment (gosonzhang) [inline inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:102]: When I use this version to verify, I found that the client did not respond to load balancing, and no longer registered Broker for consumption

**Review by gosonzhang — CHANGES_REQUESTED** [review]

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go [hunk 4]
```diff
@@ -99,36 +99,45 @@ func (h *heartbeatManager) consumerHB2Master() {
...
+		errMsg := ""
+		if err != nil {
+			if e, ok := err.(*errs.Error); !ok {
+				log.Errorf("consumer hb err %s", err.Error())
+				h.consumer.masterHBRetry++
+				h.resetMasterHeartbeat()
```
Comment (charlely) [inline inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:108]: This Error need recalc heart timer.

**Pull request merged**